modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/MultiEncodeWorkflowOperationHandler.java

```diff
@@ -0,0 +1,545 @@
+    int i = 0;
+    for (ElementProfileTagFlavor ep : elementSelectors) {
+      for (String flavor : asList(targetFlavors[i])) {
+        try {
+          ep.setTargetFlavor(flavor);
+        } catch (IllegalArgumentException e) {
+          throw new WorkflowOperationException("Target flavor " + e.getMessage());
+        }
+      }
+      if (targetFlavors.length != 1)
+        i++;
+    }
+    return elementSelectors;
+  }
+
+  private List<ElementProfileTagFlavor> populateTargetTag(List<ElementProfileTagFlavor> elementSelectors,
+          String[] targetTags) throws WorkflowOperationException {
+    if (targetTags == null)
+      return elementSelectors;
+    int i = 0;
+    for (ElementProfileTagFlavor ep : elementSelectors) {
+      try {
+        ep.setTargetTags(targetTags[i]);
+      } catch (IllegalArgumentException e) {
+        throw new WorkflowOperationException("Target flavor " + e.getMessage());
+      }
+      if (targetTags.length != 1)
+        i++;
+    }
+    return elementSelectors;
+  }
+
+  private String[] getConfigAsArray(WorkflowOperationInstance operation, String name) {
+    String sourceOption = StringUtils.trimToNull(operation.getConfiguration(name));
+    String[] options = (sourceOption != null) ? sourceOption.split(SEPARATOR) : null;
+    return (options);
+  }
+
+  /*
+   * Encode multiple tracks in a mediaPackage concurrently with different encoding profiles for each track. The encoding
+   * profiles are specified by names in a list and are the names used to tag each corresponding output. Each source
+   * track will start one operation on one worker. concurrency is achieved by running on different workers
+   *
+   * @param src The source media package
+   *
+   * @param operation the current workflow operation
+   *
+   * @return the operation result containing the updated media package
+   *
+   * @throws EncoderException if encoding fails
+   *
+   * @throws WorkflowOperationException if errors occur during processing
+   *
+   * @throws IOException if the workspace operations fail
+   *
+   * @throws NotFoundException if the workspace doesn't contain the requested file
+   */
+  private WorkflowOperationResult multiencode(MediaPackage src, WorkflowOperationInstance operation)
+          throws EncoderException, IOException, NotFoundException, MediaPackageException, WorkflowOperationException {
+    MediaPackage mediaPackage = (MediaPackage) src.clone();
+    // Check which tags have been configured
+    String[] sourceTags = getConfigAsArray(operation, "source-tags");
+    String[] sourceFlavors = getConfigAsArray(operation, "source-flavors");
+    String[] targetTags = getConfigAsArray(operation, "target-tags");
+    String[] targetFlavors = getConfigAsArray(operation, "target-flavors");
+    String tagWithProfileConfig = StringUtils.trimToNull(operation.getConfiguration("tag-with-profile"));
+    boolean tagWithProfile = tagWithProfileConfig != null && Boolean.parseBoolean(tagWithProfileConfig);
```

Comment:
`null` check is unnecessary and you might want to use [`BooleanUtils.toBoolean`](https://commons.apache.org/proper/commons-lang/javadocs/api-3.0/org/apache/commons/lang3/BooleanUtils.html#toBoolean(java.lang.String)) for user friendliness